src/daria/analysis/concentrationanalysis.py

```diff
@@ -187,11 +193,29 @@ def _extract_scalar_information(self, img: daria.Image) -> None:
             img.toGreen()
         elif self.color == "blue":
             img.toBlue()
-        elif isinstance(self.color, callable):
+        elif self.color == "hue":
+            img.toHue()
+        elif self.color == "saturation":
+            img.toSaturation()
+        elif self.color == "value":
+            img.toValue()
+        elif callable(self.color):
             img.img = self.color(img.img)
         else:
             raise ValueError(f"Mono-colored space {self.color} not supported.")
 
+    def _extract_scalar_information_after(self, img: np.ndarray) -> None:
+        """
+        Make a mono-colored image from potentially multi-colored image.
+
+        Args:
+            img (np.ndarray): image
+
+        Returns:
+            np.ndarray: monochromatic reduction of the array
+        """
+        return img
```

Comment:
This is a bit confusing to me. It looks like a function that should to something to img and return (potentially) nothing, yet it does nothing and returns img.

Reply:
The main reason is to allow modular deisgn such that with inheritance this class can be very quickly tailored to a sepcific scenario. For example: I have played a bit with this design to restrict images to the third component of the HSV color space, but only used signals which have hue in a certain interval. Instead of providing these rather custom functions, inheritance felt much simpler.